include/sys/uio_impl.h

```diff
@@ -40,10 +40,22 @@
 #define	_SYS_UIO_IMPL_H
 
 #include <sys/uio.h>
+#include <sys/sysmacros.h>
 
 extern int zfs_uiomove(void *, size_t, zfs_uio_rw_t, zfs_uio_t *);
 extern int zfs_uiocopy(void *, size_t, zfs_uio_rw_t, zfs_uio_t *, size_t *);
 extern void zfs_uioskip(zfs_uio_t *, size_t);
+extern void zfs_uio_free_dio_pages(zfs_uio_t *, zfs_uio_rw_t);
+extern int zfs_uio_get_dio_pages_alloc(zfs_uio_t *, zfs_uio_rw_t);
+
+#define	IO_ALIGNED(o, s, a)	(((o) % (a) == 0) && ((s) % (a) == 0))
```

Comment:
If the alignment is known to always be a power of 2 we can use `IS_P2ALIGNED`, such as
```C
#define	IO_ALIGNED(o, s, a) (IS_P2ALIGNED((o), (a)) && IS_P2ALIGNED((s), (a)))
```

Reply:
Big IFF though. I see `dn->dn_datablksz` is passed for the alignment in some places, so perhaps this is not appropriate after all.